gatsby-node.js

```diff
@@ -12,6 +14,78 @@ const paginationPath = (page, totalPages, prefix = `/`) => {
   }
 }
 
+const getAlbumWithTitleId = curry((title, id) => {
+  return {
+    path: mangoSlugfy(title),
+    component: path.resolve(`./src/templates/album.js`),
+    context: {
+      albumId: id,
+      slugPath: mangoSlugfy(title),
+    },
+  }
+});
+
+
+const getPaginatedPageWithContentComponentPathPrefix = curry((content, component, pathPrefix, postsPerPage, tagKeyName, prefix) => {
+  const posts = content;
+  if (!isEmpty(posts)) {
+    const numPages = Math.ceil(posts.length / postsPerPage);
+    return Array.from({ length: numPages }).map((_, i) => ({
```

Comment:
Same situation metioned above, try to keep a single return statement, storing the possible return values in a variable and then manipulating it.

In this case a return variable would be set with an empty array and if matches the criteria of the if would be populated, and then always return it.